probe_src/probe_py/analysis.py

```diff
@@ -59,59 +70,44 @@ def last(pid: int, exid: int, tid: int) -> Node:
 
     # Hook up forks/joins
     for node in list(nodes):
-        pid, exid, tid, op = node
-        if False:
-            pass
-        elif isinstance(op, CloneOp):
+        pid, exid, tid, op, node_id, op_index, _time = node
+        if isinstance(op, CloneOp):
             if op.flags & CLONE_THREAD:
                 # Spawning a thread links to the current PID and exec epoch
-                target = (pid, exid, op.child_thread_id)
+                target = (pid, _time, exid, op.child_thread_id)
             else:
-                # New process always links to exec epoch 0 and main thread
-                # THe TID of the main thread is the same as the PID
-                target = (op.child_process_id, 0, op.child_process_id)
+                target = (op.child_process_id, -1, 0, 0)
             exec_edges.append((node, first(*target)))
         elif isinstance(op, WaitOp) and op.ferrno == 0 and op.ret > 0:
             # Always wait for main thread of the last exec epoch
             if op.ferrno == 0:
-                target = (op.ret, last_exec_epoch.get(op.ret, 0), op.ret)
+                target = (op.ret, -1, last_exec_epoch.get(op.ret, 0), op.ret)
```

Comment:
@charmoniumQ In the previous version of the function, the tid being fetched was op.ret. Shouldn't it be 0, as we need the main thread of the last execution epoch? Should I change it, or is it correct as it is?

Reply:
Ah, I changed the meaning of `tid` under-the-hood. I'm now using Linux's TID directly. In Linux, the main thread has TID == PID; all the other threads have random TIDs.

I had to switch to Linux TIDs because we aren't the only one using TIDs; `waittid` returns a (Linux) TID, and we need to correlate that with one of our threads.